<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Simple Processing</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../libsources/icon.qrc">
    <normaloff>:/icon/simpleProcessing.ico</normaloff>:/icon/simpleProcessing.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QSmartGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>940</width>
      <height>610</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(89, 89, 89);</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>10</y>
      <width>301</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string>Histogram</string>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
    <widget class="QSmartLabel" name="histogram">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>256</width>
       <height>100</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(89, 89, 89);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>150</y>
      <width>301</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Simple Processing</string>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>281</width>
       <height>58</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Contrast</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Gray Res</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QSlider" name="horizontalSlider_contrast">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>50</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_contrastInfo">
            <property name="minimumSize">
             <size>
              <width>25</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>25</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="scaledContents">
             <bool>false</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QSlider" name="horizontalSlider_grayResolution">
            <property name="maximum">
             <number>8</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="pageStep">
             <number>4</number>
            </property>
            <property name="value">
             <number>8</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_grayResolutionInfo">
            <property name="minimumSize">
             <size>
              <width>25</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>25</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>8.0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>400</y>
      <width>301</width>
      <height>231</height>
     </rect>
    </property>
    <property name="title">
     <string>Preview</string>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
    <widget class="QSmartGraphicsView" name="graphicsView_preview">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>281</width>
       <height>200</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(89, 89, 89);</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>21</height>
    </rect>
   </property>
   <property name="acceptDrops">
    <bool>true</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>false</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport">
     <property name="title">
      <string>Export</string>
     </property>
     <addaction name="actionTo_PDF"/>
    </widget>
    <addaction name="actionOpen_Image"/>
    <addaction name="separator"/>
    <addaction name="actionPrint"/>
    <addaction name="menuExport"/>
    <addaction name="separator"/>
    <addaction name="actionPreference"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <widget class="QMenu" name="menuImage_Effects">
     <property name="title">
      <string>Image Effects</string>
     </property>
     <widget class="QMenu" name="menuGray_scale">
      <property name="title">
       <string>Gray Scale</string>
      </property>
      <addaction name="actionGrayscale_average"/>
      <addaction name="actionGrayscale_luminosity"/>
     </widget>
     <widget class="QMenu" name="menuNoise_Reduction">
      <property name="title">
       <string>Noise Reduction</string>
      </property>
      <addaction name="actionMean_Filter"/>
      <addaction name="actionMedian_Filter"/>
     </widget>
     <widget class="QMenu" name="menuFind_Contour">
      <property name="title">
       <string>Find Contour</string>
      </property>
      <addaction name="actionFuzzy_Logic"/>
      <addaction name="actionLaplacian_Filter"/>
      <addaction name="actionSobel_Filter"/>
     </widget>
     <addaction name="actionBlur"/>
     <addaction name="actionContrast"/>
     <addaction name="menuFind_Contour"/>
     <addaction name="actionGray_Resolution_Scale"/>
     <addaction name="menuGray_scale"/>
     <addaction name="actionBi_histogram_Equalization"/>
     <addaction name="actionHistogram_Equalization"/>
     <addaction name="menuNoise_Reduction"/>
     <addaction name="actionThreshold"/>
     <addaction name="separator"/>
    </widget>
    <addaction name="actionBack"/>
    <addaction name="actionForward"/>
    <addaction name="separator"/>
    <addaction name="menuImage_Effects"/>
    <addaction name="actionResize"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionColor_Splitter"/>
    <addaction name="actionCustom_Filter"/>
    <addaction name="actionFourier_Transform"/>
    <addaction name="actionImage_Subtractor"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionWritten_by_KKyang"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuTools"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpenImage"/>
   <addaction name="separator"/>
   <addaction name="actionSpectralFilteringToolMenubar"/>
   <addaction name="separator"/>
   <addaction name="actionOptions"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionOpen_Image">
   <property name="text">
    <string>Open Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionGrayscale_average">
   <property name="text">
    <string>Average</string>
   </property>
  </action>
  <action name="actionGrayscale_luminosity">
   <property name="text">
    <string>Luminosity</string>
   </property>
  </action>
  <action name="actionResize">
   <property name="text">
    <string>Resize</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionBack">
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionForward">
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionThreshold">
   <property name="text">
    <string>Threshold</string>
   </property>
  </action>
  <action name="actionHistogram_Equalization">
   <property name="text">
    <string>Histogram Equalization</string>
   </property>
  </action>
  <action name="actionContrast">
   <property name="text">
    <string>Contrast</string>
   </property>
  </action>
  <action name="actionBlur">
   <property name="text">
    <string>Blur</string>
   </property>
  </action>
  <action name="actionPreference">
   <property name="text">
    <string>Preference</string>
   </property>
  </action>
  <action name="actionOpenImage">
   <property name="icon">
    <iconset resource="../libsources/octicons/octicons.qrc">
     <normaloff>:/light/file-media-light.svg</normaloff>:/light/file-media-light.svg</iconset>
   </property>
   <property name="text">
    <string>openImage</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="icon">
    <iconset resource="../libsources/octicons/octicons.qrc">
     <normaloff>:/light/gear-light.svg</normaloff>:/light/gear-light.svg</iconset>
   </property>
   <property name="text">
    <string>options</string>
   </property>
  </action>
  <action name="actionGray_Resolution_Scale">
   <property name="text">
    <string>Gray Resolution Scale</string>
   </property>
  </action>
  <action name="actionMedian_Filter">
   <property name="text">
    <string>Median Filter</string>
   </property>
  </action>
  <action name="actionMean_Filter">
   <property name="text">
    <string>Mean Filter</string>
   </property>
  </action>
  <action name="actionFuzzy_Logic">
   <property name="text">
    <string>Fuzzy Logic</string>
   </property>
  </action>
  <action name="actionLaplacian_Filter">
   <property name="text">
    <string>Laplacian Filter</string>
   </property>
  </action>
  <action name="actionCustom_Filter">
   <property name="text">
    <string>Custom Filter</string>
   </property>
  </action>
  <action name="actionImage_Subtractor">
   <property name="text">
    <string>Image Subtractor</string>
   </property>
  </action>
  <action name="actionSobel_Filter">
   <property name="text">
    <string>Sobel Filter</string>
   </property>
  </action>
  <action name="actionPrint">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Print</string>
   </property>
  </action>
  <action name="actionTo_PDF">
   <property name="text">
    <string>to PDF</string>
   </property>
  </action>
  <action name="actionWritten_by_KKyang">
   <property name="text">
    <string>Written by KKyang</string>
   </property>
  </action>
  <action name="actionFourier_Transform">
   <property name="text">
    <string>Fourier Transform</string>
   </property>
  </action>
  <action name="actionSpectralFilteringToolMenubar">
   <property name="text">
    <string>spectralFilteringTool</string>
   </property>
  </action>
  <action name="actionBi_histogram_Equalization">
   <property name="text">
    <string>Bi-histogram Equalization</string>
   </property>
  </action>
  <action name="actionColor_Splitter">
   <property name="text">
    <string>Color Splitter</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QSmartGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header location="global">qsmartgraphicsview.h</header>
  </customwidget>
  <customwidget>
   <class>QSmartLabel</class>
   <extends>QLabel</extends>
   <header>qsmartlabel.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../libsources/octicons/octicons.qrc"/>
  <include location="../libsources/icon.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>359</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
